File: Sort/QuickSort/215_kth _largest_element/main.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        def partition(nums, left, right):
            pivot = random.randint(left, right - 1)
            nums[pivot], nums[left] = nums[left], nums[pivot]
            low_idx = pivot_point = left
            for i in range(left + 1, right):
                if nums[i] < nums[pivot_point]:
                    low_idx += 1
                    if low_idx != i:
                        nums[i], nums[low_idx] = nums[low_idx], nums[i]
            nums[low_idx], nums[pivot_point] = nums[pivot_point], nums[low_idx]
            return low_idx

        def quicksort(nums, left, right):
            if left == right:
                return
            pivot = partition(nums, left, right)
            quicksort(nums, left, pivot)
            quicksort(nums, pivot + 1, right)

        quicksort(nums, 0, len(nums))
        return nums[-k]

File: Sort/QuickSort/215_kth _largest_element/main.go
package main

import "math/rand"

func findKthLargest(nums []int, k int) int {
	var quickSort func([]int, int, int)
	var partition func([]int, int, int) int

	partition = func(nums []int, left int, right int) int {
		pivot := rand.Intn(right-left) + left
		nums[left], nums[pivot] = nums[pivot], nums[left]
		low_index := left
		pivot_point := left
		for i := left + 1; i < right; i++ {
			if nums[i] < nums[pivot_point] {
				low_index++
				if low_index != i {
					nums[low_index], nums[i] = nums[i], nums[low_index]
				}
			}
		}
		nums[low_index], nums[pivot_point] = nums[pivot_point], nums[low_index]
		return low_index
	}

	quickSort = func(nums []int, left, right int) {
		if left == right {
			return
		}
		pivot := partition(nums, left, right)
		quickSort(nums, left, pivot)
		quickSort(nums, pivot+1, right)
	}

	quickSort(nums, 0, len(nums))
	fmt.Print(nums)

	return nums[len(nums)-k]
}
